project/Timeslot.cs

```diff
@@ -104,28 +108,33 @@ public Exam Exam
         get => _exam;
         set
         {
-            if (value != null && value.Timeslot != null && value.Timeslot != this)
-                throw new InvalidOperationException(
-                    "The exam is already assigned to another timeslot."
-                );
+            if (_exam == value) return; // Avoid redundant operations
+            if (_exam != null && value != null && _exam != Defaults.DEFAULT_EXAM && _exam != value)
+                throw new InvalidOperationException("Cannot overwrite an existing valid exam.");
+            
             _exam = value;
             if (value != null && value.Timeslot != this)
+            {
                 value.Timeslot = this;
+            }
         }
     }
 
     public void AddExam(Exam exam)
     {
-        if (_exam != null)
+        if (_exam != null && _exam != exam)
```

Comment:
```suggestion
        if (_exam != Defaults.DEFAULT_EXAM)
```

This is the best way, since technically our default objects are our "null" values..